### PR TITLE
SNS replaced with Redis

### PR DESCRIPTION
**Pull request by mrtcode**

### CODE REVIEW

File: model/NotifierObserver.inc.php [hunk 1]
```diff
@@ -152,17 +143,22 @@ public static function notify($event, $type, $ids, $extraData) {
 						$message[$key] = $val;
 					}
 				}
-				self::send($message);
+				foreach (self::$messageReceivers as $receiver) {
+					$receiver(
+						$apiKey,
+						json_encode($message, JSON_UNESCAPED_SLASHES)
+					);
+				}
```
Comment (dstillman) [inline model/NotifierObserver.inc.php]: Can we extract this out into a separate function so we can use the same thing for the call with `$topic` above without repeating the `json_encode($message, JSON_UNESCAPED_SLASHES)` stuff?

File: model/NotifierObserver.inc.php [hunk 2]
```diff
@@ -152,17 +143,22 @@ public static function notify($event, $type, $ids, $extraData) {
 						$message[$key] = $val;
 					}
 				}
-				self::send($message);
+				foreach (self::$messageReceivers as $receiver) {
+					$receiver(
+						$apiKey,
```
Comment (dstillman) [inline model/NotifierObserver.inc.php]: Let's take this opportunity to start hashing these on both sides — there's no reason to send raw API keys through the queue.

**Comment by mrtcode** [comment]
Added key ID to Zotero_Key::toJSON. This is maybe a bit cleaner than returning it in header. If we'll think this is the right way to do this, I will update the rest (tests, etc.)

**Comment by dstillman** [comment]
I don't think it needs to be visible in the JSON by default — not because it's sensitive, but just because we'll be the only consumers and there's no need to expose it unnecessarily or clutter the output. We could just test for `!empty($_GET['showid'])` or something like that before showing. (Normally we'd send this through `$queryParams`, but probably not worth the effort here.)

**Pull request merged**